python/gui/MaaS/templates/maas/dataset_management.html

```diff
@@ -0,0 +1,797 @@
+            buildDetailsTableRow(table, "Created", dataset_json["create_on"]);
+            buildDetailsTableRow(table, "Last Updated", dataset_json["last_updated"]);
+
+            // TODO: add something for dataset size
+            // TODO: add something for number of files.
+            // TODO: add something for indicating if data aligns properly with configured domain
+
+            // Put spacer row in to indicate the start of domain details
+            let domain_header_row = document.createElement('tr');
+            let colCell = document.createElement('th');
+            colCell.colSpan = 2;
+            colCell.style.textAlign = 'left';
+            colCell.style.paddingTop = '20px';
+            colCell.style.paddingLeft = '5%';
+            colCell.style.fontStyle = 'italic';
+            colCell.style.fontSize = '18px';
+            colCell.appendChild(document.createTextNode("Data Domain"));
+            domain_header_row.appendChild(colCell);
+            table.appendChild(domain_header_row);
+
+            // Then iteratively add domain details rows
+            let c_restrict;
+            for (let c_i = 0; c_i < dataset_json["data_domain"]["continuous"].length; c_i++) {
+                c_restrict = dataset_json["data_domain"]["continuous"][c_i];
+                buildDetailsTableRow(table, c_restrict["variable"], c_restrict["begin"] + " to " + c_restrict["end"]);
+            }
+            let d_restrict;
+            for (let d_i = 0; d_i < dataset_json["data_domain"]["discrete"].length; d_i++) {
+                d_restrict = dataset_json["data_domain"]["discrete"][d_i];
+                buildDetailsTableRow(table, d_restrict["variable"], d_restrict["values"]);
+            }
+        }
+
+        function makeByteSizesReadable(bytes, decimal_places = 2) {
+            if (bytes <= 0) {
+                return "0 B";
+            }
+            let thresh = 1024;
+            let decimals = decimal_places < 0 ? 0 : decimal_places;
+            let units = ['B', 'KB', 'MB', 'GB', 'TB'];
+            let level = Math.floor(Math.log(bytes) / Math.log(thresh))
+
+            return `${parseFloat((bytes / Math.pow(thresh, level)).toFixed(decimals))} ${units[level]}`
+        }
+
+        function clickDownloadDatasetFile(dataset_name, file_name) {
+            $.ajax({
+                url: "/dataset-api",
+                type: 'get',
+                data: {"request_type": "download_file", "dataset_name": dataset_name, "item_name": file_name},
+                success: function(response) {
+                    let x = 1;
+                },
+                error: function(response) {
+                    alert('Received an error downloading file ' + file_name + ' from dataset ' + dataset_name);
+                    console.log(response);
+                }
+            });
+        }
+
+        function ajaxBuildDatasetFilesTable(dataset_name, contents_json, table) {
```

Comment:
It might be easier to take advantage of a [template tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) here instead of programmatic element creation. That'd help reduce possible nesting errors and make things a little easier to read (since you'd be reading markup instead of js).